### ISSUE TITLE
List of all MQTT Topics

### ISSUE THREAD

**Issue by Daniel-dev22** [jgyates/genmon#893]
Is there a list of all MQTT topics genmon supports for both reporting and payloads for commands?

**Comment by jgyates** [comment]
the easiest way is to just dump the topics. The [wiki page for this add on](https://github.com/jgyates/genmon/wiki/1----Software-Overview#genmqttpy-optional) shows hot to use the mosquitto_sub program (which is included with the mospuitto MQTT broker):

       mosquitto_sub -h localhost -t generator/# -u username -P password -F %t:%p

Additionally, the topics are generated dynamically, however you can easily get a feel for the names of the topics by looking at the web interface. The Status, Maintenance, Outage, and Monitor pages of the web interface show data, presented in a structured form. This form is parsed directly into MQTT topics and "generator" is used for the root topic. For example if you look at the Monitor page, the first entry is 

Monitor :
    Generator Monitor Stats :
        Monitor Health : OK

The MQTT topic would be

     /generator/Monitor/Generator Monitor Stats/Monitor Health/"OK"

Write commands allow any existing command that genmon accepts. Details on write commands are [here](https://github.com/jgyates/genmon/wiki/1----Software-Overview#genmonpy-required)

The reason for the dynamic nature of the design is that the displayed parameters differ for each generator supported. This approach allows all generators supported to use MQTT, including ones that use the custom interface allowed by the software.

**Comment by jgyates** [comment]
I updated the wiki with some of the above info that was not there already.

https://github.com/jgyates/genmon/wiki/1----Software-Overview#genmqttpy-optional

**Comment by jgyates** [comment]
@Daniel-dev22 

I don't think I understand your question. Let me know if this does not answer your question. The wiki lists commands you can issue to the genmon software. One of the commands is the "setremote" command. The "setremote" command has arguments. One of the arguments is "start" and another is "stop" and a third is "startstransfer"

The first sentence of the wiki paragraph can be summarized by:
"setremote start" should **not** be followed by a "setremote startstranfer". "startremote start" should be fllowed by teh "setremote stop" command".

To give it more detail you can issue the "start" command, wait for a period of time and then issue the "stop" command, but you the controller will ignore the "starttransfer" command unless you stop the generator first.

The 2nd sentence just states how the generator acts.

**Comment by jgyates** [comment]
I think the thing that may be the source of the confusion is that the starttranfer comands does two things. The starttransfer command will start (crank) then generator **and then** signal to the transfer switch to engage. The command will be ignored (i.e. will not engage the transfer switch) if the generator is already running (hence the reason you need to make sure it is stopped first).

The key thing is that the starttransfer command actually cranks the engine, then, once the engine is running it will signal to the transfer switch.

To summarize there are three main commands:

setremote start  - this will start the generator
setremote starttransfer - this will start then engine and then signal the transfer switch to engage
setremote stop. - this will stop the generator engine when it was started by one of the two previous commands

The starttransfer command will not engage the transfer switch if the engine is running before the starttransfer command is issued.
